### PR TITLE
[e2e-tests] tests for plugin insert buttons

### PR DESCRIPTION
**Pull request by noamtwix**
- Insert plugins tests, check that plugin is created successfully from Side & Footer toolbars.

- Insert plugins native upload tests for media plugins (File upload, Gallery & Image).

### CODE REVIEW

File: e2e/tests/insertPlugins.e2e.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    parent: { title },
+  },
+}) =>
+  cy.eyesOpen({
+    appName: 'Insert Plugins',
+    testName: title,
+    browser: DEFAULT_DESKTOP_BROWSERS,
+  });
+
+const contextSettings = () => {
```
Comment (shaulgo) [inline e2e/tests/insertPlugins.e2e.js:103]: no need for this

File: e2e/tests/insertPlugins.e2e.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    cy.switchToDesktop();
+  });
+
+  afterEach(() => cy.matchContentSnapshot());
+
+  after(() => cy.eyesClose());
+};
+
+describe('insert plugins tests', () => {
+  context('side toolbar', () => {
```
Comment (shaulgo) [inline e2e/tests/insertPlugins.e2e.js:118]: add the before, beforeEach... here.

**Comment by oded-soffrin** [comment]
Did we gave up on this PR ? @shaulgo @maimonav

**Comment by shaulgo** [comment]
Aviv is taking it after the collapsible list.

File: e2e/tests/insertPlugins.e2e.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    cy.wait(1500);
+
+    cy.eyesCheckWindow(this.test.title);
+  });
+};
+
+const testNativeUploadMediaPlugin = toolbar => ([plugin, pluginButtonName]) => {
+  return it(`should upload native ${plugin?.toLocaleLowerCase()}`, function() {
+    const testAppConfig = {
+      ...useUploadConfig({ isNativeUpload: true }),
```
[the remark this answers is not in the document]
Comment (maimonav) [reply e2e/tests/insertPlugins.e2e.js:69]: done.

File: e2e/test-env/src/client/RicosTestApp/RicosTestApp.jsx [hunk 4]
```diff
@@ -32,6 +38,20 @@ class RicosTestApp extends PureComponent {
 
     const { contentState, onEditorChange, locale, isMobile, testAppConfig = {} } = this.props;
     const { addPluginMenuConfig, footerToolbarConfig } = testAppConfig.toolbarConfig || {};
+    const isNativeUpload = testAppConfig?.uploadConfig?.isNativeUpload;
```
[the remark this answers is not in the document]
Comment (maimonav) [reply e2e/test-env/src/client/RicosTestApp/RicosTestApp.jsx:52]: done.

File: e2e/test-env/src/client/RicosTestApp/RicosTestApp.jsx [hunk 5]
```diff
@@ -32,6 +38,20 @@ class RicosTestApp extends PureComponent {
...
+      ? {
+          'wix-draft-plugin-file-upload': {
+            handleFileSelection: mockTestFileUpload,
+          },
+        }
+      : {
+          'wix-draft-plugin-file-upload': {
+            onFileSelected: mockTestFileNativeUpload,
+          },
+        };
```
[the remark this answers is not in the document]
Comment (maimonav) [reply e2e/test-env/src/client/RicosTestApp/RicosTestApp.jsx:63]: done.

File: e2e/tests/insertPlugins.e2e.js (hunk 2 above)
Comment (maimonav) [reply e2e/tests/insertPlugins.e2e.js:118]: done.

File: e2e/cypress/support/commands.js [hunk 6]
```diff
@@ -592,6 +592,23 @@ Cypress.Commands.add('triggerLinkPreviewViewerUpdate', () => {
     .should('be.visible');
 });
 
+Cypress.Commands.add('insertPlugin', (toolbar, pluginInsertButtonName) => {
```
[the remark this answers is not in the document]
Comment (maimonav) [reply e2e/cypress/support/commands.js:635]: done.

File: e2e/tests/insertPlugins.e2e.js (hunk 1 above)
Comment (maimonav) [reply e2e/tests/insertPlugins.e2e.js:103]: done.

File: e2e/tests/insertPlugins.e2e.js [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+} from '../cypress/dataHooks';
+import { DEFAULT_DESKTOP_BROWSERS } from './settings';
+import { useUploadConfig } from '../cypress/testAppConfig';
+
+const LINKS = {
+  YOUTUBE: 'https://www.youtube.com/watch?v=whbidPR4nVA',
+  SOUNDCLOUD: 'https://soundcloud.com/martingarrix/martin-garrix-animals-original',
+};
+
+const modalHandler = (INPUT_BUTTON, ADD_BUTTON, LINK) => {
```
Comment (shaulgo) [inline e2e/tests/insertPlugins.e2e.js:20]:
```suggestion
const modalHandler = (INPUT, ADD_BUTTON, LINK) => {
```

File: e2e/tests/insertPlugins.e2e.js [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+  cy.get(`[data-hook=${INPUT_BUTTON}]`).type(LINK);
+  cy.get(`[data-hook=${ADD_BUTTON}]`).click();
+  cy.wait(3000); //DO NOT REMOVE - fix flakiness
+};
+
+const additionalCommands = {
+  VIDEO: () => {
+    modalHandler(VIDEO_PLUGIN.INPUT, VIDEO_PLUGIN.ADD, LINKS.YOUTUBE);
```
Comment (shaulgo) [inline e2e/tests/insertPlugins.e2e.js:28]: try to maybe do it like this
```suggestion
    modalHandler(LINKS.YOUTUBE);
```

File: e2e/tests/insertPlugins.e2e.js [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+  beforeEach('load editor', () => {
+    cy.switchToDesktop();
+  });
+
+  afterEach(() => cy.matchContentSnapshot());
+
+  after(() => cy.eyesClose());
+
+  context('side toolbar', () => {
+    Object.entries(STATIC_TOOLBAR_BUTTONS_WITHOUT_EMBED).map(testInsertPlugin(TOOLBARS.SIDE));
```
Comment (shaulgo) [inline e2e/tests/insertPlugins.e2e.js:116]:
```suggestion
    Object.entries(STATIC_TOOLBAR_BUTTONS_WITHOUT_EMBED).forEach(testInsertPlugin(TOOLBARS.SIDE));
```

File: e2e/tests/insertPlugins.e2e.js [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+    Object.entries(STATIC_TOOLBAR_BUTTONS_MEDIA).map(testNativeUploadMediaPlugin(TOOLBARS.SIDE));
```
Comment (shaulgo) [inline e2e/tests/insertPlugins.e2e.js:117]:
```suggestion
    Object.entries(STATIC_TOOLBAR_BUTTONS_MEDIA).forEach(testNativeUploadMediaPlugin(TOOLBARS.SIDE));
```

Comment (shaulgo) [reply e2e/tests/insertPlugins.e2e.js:117]: and the rest

File: e2e/tests/insertPlugins.e2e.js [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+    additionalCommands[plugin]?.();
+
+    cy.wait(1500);
+
+    cy.eyesCheckWindow(this.test.title);
+  });
+};
+
+const testNativeUploadMediaPlugin = toolbar => ([plugin, pluginButtonName]) => {
+  return it(`should upload native ${plugin?.toLocaleLowerCase()}`, function() {
```
Comment (shaulgo) [inline e2e/tests/insertPlugins.e2e.js:66]:
```suggestion
 it(`should upload native ${plugin?.toLocaleLowerCase()}`, function() {
```

File: e2e/tests/insertPlugins.e2e.js [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+  },
+  // GIPHY: () => {
+  //   cy.get(`[data-hook=${GIPHY_PLUGIN.UPLOAD_MODAL}]`);
+  //   cy.get(`[role=button][tabindex=0]:first`).click();
+  // },
+  // ADSENSE: () => {},
+};
+
+const testInsertPlugin = toolbar => ([plugin, pluginButtonName]) => {
+  return it(`should insert ${plugin?.toLocaleLowerCase()}`, function() {
```
Comment (shaulgo) [inline e2e/tests/insertPlugins.e2e.js:52]:
```suggestion
 it(`should insert ${plugin?.toLocaleLowerCase()}`, function() {
```

File: examples/main/shared/utils/fileUploadUtil.js [hunk 13]
```diff
@@ -113,3 +113,80 @@ export const getVideoToUpload = (url, thumbnailUrl) => {
   // If relative URL is provided, a function 'getVideoUrl' will be invoked to form a full URL.
   return videoWithRelativeUrl;
 };
+
+//////////////////////////////////////////TESTS PURPOSES//////////////////////////////////////////
```
Comment (shaulgo) [inline examples/main/shared/utils/fileUploadUtil.js:117]:
```suggestion
//////////////////////////////////////////FOR TESTS//////////////////////////////////////////
```

File: examples/main/shared/utils/fileUploadUtil.js [hunk 14]
```diff
@@ -113,3 +113,80 @@ export const getVideoToUpload = (url, thumbnailUrl) => {
...
+  data.push({
+    name,
+    type,
+    url: 'http://file-examples.com/wp-content/uploads/2017/10/file-sample_150kB.pdf',
+  });
+
+  setTimeout(() => updateEntity({ data }), 200);
+};
+
+//////////////////////////////////////////TESTS PURPOSES//////////////////////////////////////////
```
Comment (shaulgo) [inline examples/main/shared/utils/fileUploadUtil.js:192]:
```suggestion
//////////////////////////////////////////FOR TESTS//////////////////////////////////////////
```

**Review by shaulgo — APPROVED** [review]

File: e2e/tests/insertPlugins.e2e.js (hunk 7 above)
Comment (maimonav) [reply e2e/tests/insertPlugins.e2e.js:20]: done.

File: e2e/tests/insertPlugins.e2e.js (hunk 12 above)
Comment (maimonav) [reply e2e/tests/insertPlugins.e2e.js:52]: done.

File: e2e/tests/insertPlugins.e2e.js (hunk 11 above)
Comment (maimonav) [reply e2e/tests/insertPlugins.e2e.js:66]: done.

File: e2e/tests/insertPlugins.e2e.js (hunk 9 above)
Comment (maimonav) [reply e2e/tests/insertPlugins.e2e.js:116]: done.

File: e2e/tests/insertPlugins.e2e.js (hunk 10 above)
Comment (maimonav) [reply e2e/tests/insertPlugins.e2e.js:117]: done.

File: examples/main/shared/utils/fileUploadUtil.js (hunk 13 above)
Comment (maimonav) [reply examples/main/shared/utils/fileUploadUtil.js:117]: done.

File: examples/main/shared/utils/fileUploadUtil.js (hunk 14 above)
Comment (maimonav) [reply examples/main/shared/utils/fileUploadUtil.js:192]: done.

File: e2e/tests/insertPlugins.e2e.js (hunk 8 above)
Comment (maimonav) [reply e2e/tests/insertPlugins.e2e.js:28]: I'll keep it like that for now.
If we'll see in the future that it needs to be changed - I'll make a more generic implementation.

**Pull request merged**